packages/p2p/src/pages/buy-sell/buy-sell-form.jsx

```diff
@@ -110,19 +110,21 @@ const BuySellForm = props => {
         const receive_amount = input_amount * calculated_rate;
         buy_sell_store.setReceiveAmount(receive_amount);
         // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [input_amount, effective_rate]);
+    }, [input_amount, calculated_rate]);
 
+    // This is to prevent the rate from changing when the MarketRateChangeErrorModal is shown on mobile and counterparty
+    // changes the rate. This is to ensure that the rate is not changed when the user is in the middle of placing an order.
     React.useEffect(() => {
-        if (exchange_rate !== previous_rate && rate_type === ad_type.FLOAT) {
-            setPreviousRate(exchange_rate);
-            showModal({
-                key: 'RateChangeModal',
-                props: {
-                    currency: buy_sell_store.local_currency,
-                },
-            });
+        if (isMobile() && has_rate_changed && current_effective_rate !== effective_rate) {
+            setChangedRate(effective_rate);
         }
-    }, [buy_sell_store.local_currency, exchange_rate, previous_rate, rate_type, showModal]);
+    }, [current_effective_rate, effective_rate, has_rate_changed]);
+
+    React.useEffect(() => {
+        if (current_effective_rate !== effective_rate && current_effective_rate !== 0) setHasRateChanged(true);
+    }, [current_effective_rate, effective_rate, setHasRateChanged]);
```

Comment:
what's the difference between these useEffects? can't we use the same for desktop and mobile?

Reply:
hmmmm we could, i need to check if its possible or not 👍 ill check back with you on this

Reply:
ah i can't combine the two, reason is because the second useEffect is used to solely show the MarketRateChangedModal and then the first useEffect, because on mobile the MarketRateChangeModal is stacked on top of the BuySellModal, if I don't have this check, and there is any change the to market rate, it will change in the background and not lock the effective rate value at the time of when the MarketRateModal is open. So this is there to prevent it. This is only on mobile